internal/peer/common/peerclient.go

```diff
@@ -148,25 +137,30 @@ func GetEndorserClient(address, tlsRootCertFile string) (pb.EndorserClient, erro
 
 // GetCertificate returns the client's TLS certificate
 func GetCertificate() (tls.Certificate, error) {
-	peerClient, err := NewPeerClientFromEnv()
+	if !viper.GetBool("peer.tls.clientAuthRequired") {
+		return tls.Certificate{}, nil
+	}
+
+	key, certificate, err := getClientAuthInfoFromEnv("peer")
 	if err != nil {
 		return tls.Certificate{}, err
 	}
-	return peerClient.Certificate(), nil
+	if key == nil || certificate == nil {
+		return tls.Certificate{}, errors.New("both Key and Certificate are required when using mutual TLS")
+	}
```

Comment:
I was going to request that we update the error message here to mention the exact names of the config parameters in the error message, but looking closer, it seems like we'll never reach this code. `getClientAuthInfoFromEnv()` only returns nils when the error is not nil. If we're trying to check for empty files here, `ioutil.ReadFile` seems to return a zero-length byte array in that case.

Reply:
@wlahti good point. I have removed the checking because `tls.X509KeyPair` will return an error for empty key/cert. Also added unit tests for GetClientCertificate.